Lib/Src/cpp_link.cpp

```diff
@@ -15,24 +17,46 @@ extern "C"
 {
 #endif
 
+#define ADC_CHANNELS 14
+
     void cpp_main()
     {
         CUartCom uart_for_errors(&huart2);
 
         CLog log_main(&uart_for_errors, "Main");
         log_main.setLogLevel(CLog::LOG_INFO);
 
+        CAdc adc_1(&hadc, ADC_CHANNELS);
```

Comment:
Right, so this works because you are creating the class within cpp_main(). If someone was to use your class and create it as a global variable, it'd fail. Test it and see.